clearnode/session_key.go

```diff
@@ -178,13 +183,18 @@ func GetActiveSessionKeysByWallet(db *gorm.DB, walletAddress string, listOpts *L
 	return sessionKeys, nil
 }
 
-// GetSessionKey retrieves a specific session key
+// GetSessionKey retrieves a specific session key and validates its expiration
```

Comment:
I think we should signal that a validation is happening inside in the function name itself, like `GetNonExpiredSessionKey`. Or separate "get" and "check" by creating a separate `validateSessionKeyNotExpired(...)` function for the latter.